backend/inspirehep/matcher/validators.py

```diff
@@ -22,3 +22,20 @@ def authors_validator(author, result):
     return bool(
         record_author_id_schema_value_pairs & result_author_id_schema_value_pairs
     )
+
+
+def collaboration_validator(author, result):
+    return author.get("collaborations") == get_value(
+        result, "_source.collaborations.value", []
+    )
+
+
+def affiliations_validator(author, result):
+    authors_affiliations = author.get("affiliations", [])
```

Comment:
better make this a `set` from the start, instead of rebuilding it inside the loop.